project_milestone/views/project_task.xml

```diff
@@ -0,0 +1,75 @@
+<odoo>
+  <!-- List View -->
+  <record model="ir.ui.view" id="view_task_enhancement_milestone_tree" >
+    <field name="name">project.task.enhancement.milestone.tree</field>
+    <field name="model">project.task</field>
+    <field name="inherit_id" ref="project.view_task_tree2"/>
+    <field eval="2" name="priority"/>
+    <field name="arch" type="xml">
+      <field name="user_id" position="after">
+        <field name="milestone_id"
+               invisible="1"/>
+      </field>
+    </field>
+  </record>
+
+  <!-- Form View -->
+  <record model="ir.ui.view" id="view_task_enhancement_milestone_form" >
+    <field name="name">project.task.enhancement.milestone.form</field>
+    <field name="model">project.task</field>
+    <field name="inherit_id" ref="project.view_task_form2"/>
+    <field eval="2" name="priority"/>
+    <field name="arch" type="xml">
+      <data>
+        <xpath expr="//field[@name='user_id']" position="after">
+          <field name="use_milestones"
+                 invisible="1"/>
+          <field name="milestone_id"
+                 domain="[('project_id', '=', project_id)]"
+                 widget="selection"
+                 attrs="{'invisible':[('use_milestones','=', False)]}"/>
+        </xpath>
+      </data>
+    </field>
+  </record>
+
+  <!-- Kanban View -->
+  <record model="ir.ui.view" id="view_task_enhancement_milestone_kanban">
+    <field name="name">project.task.enhancement.milestone.kanban</field>
+    <field name="model">project.task</field>
+    <field name="inherit_id" ref="project.view_task_kanban"/>
+    <field name="arch" type="xml">
+      <data>
+        <xpath expr="//div[hasclass('o_kanban_record_body')]" position="after">
+          <span class="o_value"><field name="milestone_id"/></span>
+        </xpath>
+      </data>
+    </field>
+  </record>
+
+  <!-- Gantt View -->
+  <record model="ir.ui.view" id="view_task_enhancement_gantt">
+    <field name="name">project.task.enhancement.milestone.gantt</field>
+    <field name="model">project.task</field>
+    <field name="arch" type="xml">
+      <gantt date_start="date_start" date_stop="date_end" progress="progress">
+      </gantt>
+    </field>
+  </record>
+
+  <!-- Search View -->
+  <record model="ir.ui.view" id="view_task_enhancement_search_milestone_form">
+    <field name="name">project.task.enhancement.milestone.search.form</field>
+    <field name="model">project.task</field>
+    <field name="inherit_id" ref="project.view_task_search_form"/>
+    <field name="arch" type="xml">
+      <filter name="inactive" position="before">

```

Comment:
Maybe it is better with position after <filter string="Company" name="company"

Reply:
Good suggestion, updated and it looks good.